src/org/usfirst/frc/team2706/robot/commands/bling/BlingController.java

```diff
@@ -0,0 +1,138 @@
+import org.usfirst.frc.team2706.robot.subsystems.Bling;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Timer;
+
+public class BlingController extends Command {
+
+    private Bling blingSystem;
+    private BlingPattern currentPattern;
+    private double startTime = 0;
+    
+    public static final int AUTONOMOUS_PERIOD = 0;
+    public static final int TELEOP_WITHOUT_CLIMB = 1;
+    public static final int CLIMBING_PERIOD = 2; 
+    
+    boolean useMatchTime = false;
+    
+    HashMap<Integer, ArrayList<BlingPattern>> commands;
+    
+    public BlingController() {
+        requires(Robot.blingSystem);
+        
+        blingSystem = Robot.blingSystem;
+        commands = new HashMap<Integer, ArrayList<BlingPattern>>() {
+            private static final long serialVersionUID = 1L;
+
+            {
+                put(AUTONOMOUS_PERIOD, new ArrayList<BlingPattern>());
+                put(CLIMBING_PERIOD, new ArrayList<BlingPattern>());
+                put(TELEOP_WITHOUT_CLIMB, new ArrayList<BlingPattern>());
+            }
+        };
+        
+        System.out.println("Going into bling controller"); // TODO remove
+        /* Make and add the bling patterns. 
+         * They need to be created in order of highest to lowest priority.
+         * 
+         * Since patterns from different periods won't run at the same time, you only have to 
+         * make sure you put patterns from the same period in proper order.
+         */
+        Add(new IntakeSignaller());
+        
+        // Do blank as a last priority
+        Add(new Blank());
+    }
+    
+    public void initialize() {
+        startTime = Timer.getFPGATimestamp();
+        // If it's already teleop when we start, just subtract 15 seconds to make it seem as though we're in teleop.
+        if (!DriverStation.getInstance().isOperatorControl()) startTime -= 15;
+        
+        // If the match time provided is not below 0, it's valid and we're in a game and use it.
+        useMatchTime = Timer.getMatchTime() >= 0;
+    }
+    
+    /**
+     * Adds the inputted command to the controller's queue or stuff to run.
+     * This will be run whenever the execute method of this controller is
+     * run, and if the conditions for the command to be run on the robot
+     * are met, show this command on the robot.
+     * @param commandToAdd The command to add to the controller's queue.
+     */
+    public void Add(BlingPattern commandToAdd){
+        // Add it to its proper place.
+        // Loop around all of the periods it can be in.
+        for (Integer i : commandToAdd.getPeriod()) {
+            // Add it to the periods it can be in
+            int period = i;
+            commands.get(period).add(commandToAdd);
+        }
+        
+        System.out.println("Added command : " + commandToAdd + " New list is " + commands); // TODO remove debug print
```

Comment:
Instead of removing debug prints, you could just use `Log.d("Title", "Message")`.